optimum/habana/transformers/integrations/awq.py

```diff
@@ -0,0 +1,215 @@
+        raise ValueError(
+            f"Only supported quantization backends in {AwqBackendPackingMethod.AUTOAWQ} - not recognized backend {self.backend}"
+        )
+
+    self.version = GaudiAWQLinearVersion.from_str(self.version)
+    if self.version not in [
+        GaudiAWQLinearVersion.HPU,
+        GaudiAWQLinearVersion.GEMM,
+    ]:
+        raise ValueError(
+            f"Only supported versions are in [GaudiAWQLinearVersion.HPU, GaudiAWQLinearVersion.GEMM] - not recognized version {self.version}"
+        )
+
+    if self.do_fuse and self.fuse_max_seq_len is None:
+        raise ValueError(
+            "You cannot enable fused modules without specifying a `fuse_max_seq_len`, make sure to pass a valid `fuse_max_seq_len` for your usecase"
+        )
+
+    if self.do_fuse:
+        awq_version_supports_fusing = False
+        MIN_AWQ_VERSION = "0.1.7"
+        if is_auto_awq_available():
+            awq_version_supports_fusing = version.parse(importlib.metadata.version("autoawq")) >= version.parse(
+                MIN_AWQ_VERSION
+            )
+
+        if not awq_version_supports_fusing:
+            raise ValueError(
+                f"You current version of `autoawq` does not support module fusing, please upgrade `autoawq` package to at least {MIN_AWQ_VERSION}."
+            )
+
+    if self.modules_to_not_convert is not None:
+        awq_version_supports_non_conversion = False
+        MIN_AWQ_VERSION = "0.1.8"
+        if is_auto_awq_available():
+            awq_version_supports_non_conversion = version.parse(
+                importlib.metadata.version("autoawq")
+            ) >= version.parse(MIN_AWQ_VERSION)
+
+        if not awq_version_supports_non_conversion:
+            raise ValueError(
+                f"You current version of `autoawq` does not support module quantization skipping, please upgrade `autoawq` package to at least {MIN_AWQ_VERSION}."
+            )
+
+    if self.do_fuse and self.modules_to_fuse is not None:
+        raise ValueError("You current implementation of `autoawq` does not support do_fuse and modules_to_fuse.")
+
+
+def gaudi_replace_with_awq_linear(
+    model,
+    modules_to_not_convert=None,
+    quantization_config=None,
+    current_key_name=None,
+    has_been_replaced=False,
+) -> bool:
+    """
+    Adapted from: https://github.com/huggingface/transformers/blob/v4.45.2/src/transformers/integrations/awq.py#L90
+    - support HPU.
+    """
+    if modules_to_not_convert is None:
+        modules_to_not_convert = []
+
+    backend = quantization_config.backend
+
+    if not is_auto_awq_available():
+        raise ValueError(
+            "AWQ (either `autoawq` or `llmawq`) is not available. Please install it with `pip install autoawq` or check out the installation guide in https://github.com/mit-han-lab/llm-awq"
+        )
+
+    if backend == AwqBackendPackingMethod.AUTOAWQ and quantization_config.version == GaudiAWQLinearVersion.HPU:
```

Comment:
what will happen if `quantization_config.version` is `GaudiAWQLinearVersion.GEMM`. From the code it sounds both are acceptible

Reply:
post_init is also called when passing the quant_config. see https://huggingface.co/TheBloke/Llama-2-7B-Chat-AWQ/blob/main/quant_config.json#L5. what's why I left GEMM here.